test/unit/manipulation.js

```diff
@@ -2162,6 +2162,16 @@ testIframeWithCallback(
 	}
 );
 
+testIframeWithCallback(
+	"domManip executes scripts in iframes in the iframes' context",
+	"manipulation/scripts-context.html",
+	function( bodyElement, html, assert ) {
+		assert.expect( 1 );
+		jQuery( bodyElement ).append( html );
+		assert.ok( !window.scriptTest, "script executed in iframe context" );
```

Comment:
It would be good if we had a way of checking if it was set in the iframe context (and not just not set in the current one) but otherwise LGTM.